### PR TITLE
The class std::iterator got deprecated. 

### PR DESCRIPTION
**Pull request by afabri**
## Summary of Changes

It only served to define some nested types.

## Release Management

* Affected package(s):
* Issue(s) solved (if any): fix #0000, fix #0000,...
* Feature/Small Feature (if any):

### CODE REVIEW

**Comment by sloriot** [comment]
is it `boost::iterator` that is deprecated or `std::iterator`?

File: STL_Extension/include/CGAL/internal/boost/array_binary_tree.hpp [hunk 1]
```diff
@@ -52,12 +52,12 @@ class array_binary_tree_node {
...
     struct iterator
-        : ::boost::iterator<std::forward_iterator_tag,
-            //JT: the iterator type is "forward" and not "bidirectional",
-            //because it does not implement unary operator--
-            ArrayBinaryTreeNode,
-            difference_type, array_binary_tree_node*, ArrayBinaryTreeNode&>
     { // replace with iterator_adaptor implementation -JGS
+      typedef std::forward_iterator_tag Category;
+      typedef ArrayBinaryTreeNode value_type;
+      typedef size_type difference_type
```
Comment (sloriot) [inline STL_Extension/include/CGAL/internal/boost/array_binary_tree.hpp]: `size_type` is not defined

Comment (afabri) [reply STL_Extension/include/CGAL/internal/boost/array_binary_tree.hpp]: seven lines before.

**Comment by lrineau** [comment]
Should not this be for 4.12.1 instead of `master`?

**Pull request merged**